packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -604,9 +622,13 @@ function parseSecretName(construct: IConstruct, secretArn: string) {
       return resourceName;
     }
 
-    // Secret resource names are in the format `${secretName}-${SecretsManager suffix}`
-    // If there is no hyphen, assume no suffix was provided, and return the whole name.
-    return resourceName.substr(0, resourceName.lastIndexOf('-')) || resourceName;
+    // Secret resource names are in the format `${secretName}-${6-character SecretsManager suffix}`
+    // If there is no hyphen (or 6-character suffix) assume no suffix was provided, and return the whole name.
+    const lastHyphenIndex = resourceName.lastIndexOf('-');
+    if (lastHyphenIndex !== -1 && resourceName.substr(lastHyphenIndex + 1).length === 6) {
+      return resourceName.substr(0, lastHyphenIndex);
+    }
+    return resourceName;
```

Comment:
Can you re-write this just slightly?

```suggestion
    const hasHyphenAndSixCharSuffix = lastHyphenIndex !== -1 && resourceName.substr(lastHyphenIndex + 1).length === 6;
    return hasHyphenAndSixCharSuffix ? resourceName.substr(0, lastHyphenIndex) : resourceName;
```